Dockerfile

```diff
@@ -25,11 +25,8 @@ RUN apt-get update \
     && rm -Rf /usr/share/doc && rm -Rf /usr/share/man \
     && apt-get clean
 
-# Upgrade pip to latest version.
-RUN pip3 install --upgrade pip
-
 # Install Ansible via pip.
-RUN pip3 install $pip_packages
+RUN pip3 install --break-system-packages $pip_packages
```

Comment:
Shouldn't we install these via the package manager if they're available?

Reply:
This is to stay consistent with [cisagov/docker-debian12-ansible](https://github.com/cisagov/docker-debian12-ansible) and also with the other `geerlingguy/docker-*-ansible` images. I want to minimize the deviation from the shared parentage as much as possible. Arguably if you are doing Ansible development using these images you will also want the latest version of Ansible as well. We are only using these images for testing and do not actually perform any Ansible work in the images themselves (Ansible playbooks are run against the images from the host by `moldecule`).

Reply:
OK for now, but I think we should soon start imposing a severe restriction to never install anything via `pip` against the system Python.  If you find yourself needing to do that then you should instead use a venv or similar.

I've thought this to be a good idea for a while, and I'm looking to use the new behavior under Debian 12 as an impetus to enforce it.

Thoughts?

Reply:
I'm totally fine with that in general for our configurations. I consider this repo and [cisagov/docker-debian12-ansible] (and the soon-to-be `cisagov/docker-debian13-ansible`) as special cases as they are heavily based on an external configuration. We will eventually be rotating out [cisagov/docker-debian12-ansible] for the eventual `geerlingguy/docker-debian12-ansible` which would behave in a similar way so I am again just trying to provide consistency for this particular set of Docker images we use.

[cisagov/docker-debian12-ansible]: https://github.com/cisagov/docker-debian12-ansible